src/net/connector.rs

```diff
@@ -33,11 +41,20 @@ impl UdpConnector {
 
 impl NetworkConnector for UdpConnector {
     type Stream = UdpSender;
-    
+
     fn connect(&self, host: &str, port: u16, _: &str) -> error::Result<<Self as NetworkConnector>::Stream> {
         let udp_sock = try!(self.0.try_clone());
-        let sock_addr = try!(net::addr_from_trait((host, port)));
-        
+        let sock_addr = match try!(self.local_addr()) {
+            SocketAddr::V4(_) => SocketAddr::V4(SocketAddrV4::new(FromStr::from_str(host).unwrap(), port)),
+            SocketAddr::V6(n) => {
+                let mut addr: SocketAddrV6 = FromStr::from_str(format!("{}:{}", host, port).as_str())
```

Comment:
Oh thanks for letting me know it was working in some cases, that clears up some confusion.

Yes I currently see that hyper is technically doing it wrong. However, I saw the commit that you made for the fix and I now actually think the code you had before was fine after reading your rationale.

What was triggering a panic for me before was running a test with ```SearchResponse``` via the ```unicast``` method. If you see [here](https://github.com/icorderi/ssdp-rs/blob/master/src/message/ssdp.rs#L68-L71) we are manually calling ```connect``` and passing in the "correct" host (without brackets). In order to unify with what hyper gives us when it decides to call connect for our requests, [here](https://github.com/icorderi/ssdp-rs/blob/master/src/message/ssdp.rs#L68) we could go ahead and do ```let dst_ip_string = format!("[{}]", dst_sock_addr.ip());``` so that in ```connect``` we know that the host will always contain brackets and don't have to perform string checks.